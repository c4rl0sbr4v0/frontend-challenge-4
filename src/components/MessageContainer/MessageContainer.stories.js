import React from 'react';
import MessageContainer from './index';

export default {
  title: 'Components/Message Container',
  component: MessageContainer,
};

export const Default = () => <MessageContainer>Container</MessageContainer>;
