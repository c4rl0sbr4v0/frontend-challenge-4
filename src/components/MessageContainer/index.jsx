import React from 'react';
import PropTypes from 'prop-types';

/**
 *  MessageContainer: Render a Message Container
 */
const MessageContainer = ({ children }) => {
  return (
    <div id="landbot-messages-container" className="landbot-messages-container">
      {children}
    </div>
  );
};

export default MessageContainer;

MessageContainer.propTypes = {
  children: PropTypes.node,
};
MessageContainer.defaultProps = {
  children: '',
};
