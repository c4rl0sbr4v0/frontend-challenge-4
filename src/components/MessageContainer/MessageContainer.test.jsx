import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MessageContainer from '.';

test('Message Container Component', async () => {
  const { container } = render(
    <MessageContainer>
      <div>ey</div>
    </MessageContainer>,
  );
  expect(container.firstChild.classList.contains('landbot-messages-container')).toBe(true);
  expect(container).toMatchSnapshot();
});
