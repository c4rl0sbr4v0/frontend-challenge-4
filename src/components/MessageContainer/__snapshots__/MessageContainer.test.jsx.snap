// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Container Component 1`] = `
<div>
  <div
    class="landbot-messages-container"
    id="landbot-messages-container"
  >
    <div>
      ey
    </div>
  </div>
</div>
`;
